frontend/utils/ModalSharePage.ts

```diff
@@ -0,0 +1,128 @@
+import { defineProps } from 'vue';
+import { STelegram, STwitter, SEmail, SMastodon, SFacebookMessenger } from 'vue-socials';
+
+import type { Event } from "~/types/event";
+import type { Organization } from "~/types/organization";
+
+export const entityProps = defineProps<{
+    organization?: Organization;
+    event?: Event;
+    // group?: Group; // add group when we have it
+}>();
+
+const contentCopied = ref(false);
+
+export const checkEntityType = () => {
+    if (entityProps.event) {
+        return setData(entityProps.event);
+    } else if (entityProps.organization) {
+        return setData(entityProps.organization);
+    }
+    // add group when we have it
+};
+
+// function to grab the url to the base id of the event/org/group
```

Comment:
Minor nit for capitalization and a period for inline comments, but no stress on it! Happy to handle it :)

Reply:
Fixing this up right now. 

So strange, I thought I'd receive email notifications once ya left some comments, I'll take a look at the others you left!